paper-server/src/main/java/org/bukkit/craftbukkit/Main.java

```diff
@@ -17,6 +17,7 @@ public class Main {
     public static final java.time.Instant BOOT_TIME = java.time.Instant.now(); // Paper - track initial start time
     public static boolean useJline = true;
     public static boolean useConsole = true;
+    public static boolean printedWarning = false;
```

Comment:
this can have a better name.. maybe "checkVersionOnServerStart" or not sure... but a thing most self-explaining than printWarning

Reply:
I feel like `printedWarning` is good enough, it's just used as a ref if the later logic should run